src/main/java/seedu/address/model/person/DateOfAdmission.java

```diff
@@ -1,16 +1,19 @@
 package seedu.address.model.person;
 
+import static java.util.Objects.requireNonNull;
+
 import java.time.LocalDate;
 
+
 /**
  * Represents a Person's date of admission in the address book.
  * Guarantees: immutable;
  */
 public class DateOfAdmission {
+    //Changed validity check for date format to disallow invalid dates, as not doing so results in parser errors
     public static final String MESSAGE_CONSTRAINTS =
-            "Date of admission should be in the format of YYYY-MM-DD, and it should not be blank.";
-
-    public static final String VALIDATION_REGEX = "^\\d{4}-(0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|3[01])$";
+            "Date of admission should be in the format of YYYY-MM-DD. It should be a valid date,"
+                    + " and it should not be blank.";
```

Comment:
Please see elaboration here & verify if they are agreeable as bugs fixes:

Model: forbid field instances to have empty values because empty input is already represented by null, and if allowed, can cause storage problems.

DateParser: add isValidDate with correct error handling (not allowing invalid dates like 1990-02-99) as not doing so results in unhandled exception.

Revert previous commits that may violate feature freeze.
- case sensitivity of nric, country and email
- more detailed (and correct) error message for deleteinfo, which include the fields
- error message for update command (as already included in UG)
- error message for deleteinfoparser